ambry-store/src/main/java/com.github.ambry.store/CompactionDetails.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Contains all the details required for a compaction cycle.
+ */
+class CompactionDetails {
+  private static final byte[] ZERO_LENGTH_ARRAY = new byte[0];
+
+  private static final short VERSION_0 = 0;
+  private static final int VERSION_SIZE = 2;
+  private static final int REFERENCE_TIME_SIZE = 8;
+  private static final int SEGMENT_COUNT_SIZE = 4;
+  private static final int SEGMENT_NAME_LENGTH_SIZE = 4;
+  private static final int SWAP_SPACE_COUNT_SIZE = 4;
+
+  private final long referenceTime;
```

Comment:
In the constructor and in the getter, there is documentation. Do you think more documentation is required?